custom_components/tahoma/cover.py

```diff
@@ -24,18 +24,16 @@ async def async_setup_entry(
     data = hass.data[DOMAIN][entry.entry_id]
     coordinator = data["coordinator"]
 
-    # Includes fix for #486, which is waiting on Somfy back-end deployment
-    # Remove when DeploymentState will be returned for AwningValance
     entities = [
         Awning(device.deviceurl, coordinator)
         for device in data["platforms"].get(COVER)
-        if device.ui_class == "Awning" and device.widget != "AwningValance"
+        if "Awning" in device.definition.qualified_name
```

Comment:
Is it always in the qualified name? Previously we checked if it was part of the ui_class? Will this change be better?

Reply:
From what I saw yes it’s always in the qualified name. I’ve tested myself, and ask [here](https://github.com/iMicknl/ha-tahoma/issues/577) and [here](https://github.com/iMicknl/ha-tahoma/issues/486)

Reply:
I’ve also checked all the issues about reversed state, the qualified name was always including Awning.